src/compiler/crystal/syntax/lexer.cr

```diff
@@ -287,8 +288,17 @@ module Crystal
         line = @line_number
         column = @column_number
         char = next_char
-        if !@slash_is_regex && char == '='
+        if (@wants_def_or_macro_name || !@slash_is_regex) && char == '/'
+          case next_char
```

Comment:
Can be replaced by a smaller `if` statement

Reply:
Using the case is more about following the pattern of the rest of the lexer.
But I don't follow which is the proposed smaller `if`.

Reply:
This is fine - this just makes the lexer looks a bit more complex that it is, that's it :)
Here it would be
```crystal
if next_char == '/'
  next_char :"//="
else
  @token.type = :"//"
end
```

Reply:
I thought you ment the `if (@wants_def_or_macro_name ...`.
There are some `if` in the lexer, bust mostly `case` even single `when`-ed (?) case. I found them easier to read the lexer if all the structures are the same and not case by case based that we will end up having if your suggestions are applied.

At there is no real change the the actual code.

So `case` are kept. But thanks.